torch/distributed/tensor/_dispatch.py

```diff
@@ -390,7 +386,14 @@ def redistribute_local_args(
                         if debug_mode is not None
                         else contextlib.nullcontext()
                     )
-
+                    if not ExplicitRedistributionContext.is_redistribute_allowed(
+                        arg_spec,
+                        # pyrefly: ignore [bad-argument-type]
+                        reshard_arg_spec,
+                    ):
+                        raise RuntimeError(
+                            f"Implicit redistribution occurred while ExplicitRedistributionContext was active for {op_info.schema}"
```

Comment:
nit
```suggestion
                            f"Implicit redistribution occurred for {op_info.schema} while ExplicitRedistributionContext was active"
```
or please adjust if you want to put `{op_info.schema}` at the end of print.